packages/server/src/server.ts

```diff
@@ -204,14 +208,27 @@ export class ProsopoServer {
         commitmentId?: string,
         maxVerifiedTime = DEFAULT_MAX_VERIFIED_TIME_CACHED
     ) {
-        this.logger.info('Verifying with provider.')
+        const blockNumberString = blockNumber.toString()
+        const dappUserSignature = this.pair?.sign(blockNumberString)
+        if (!dappUserSignature) {
+            throw new Error('Failed to sign the block number')
```

Comment:
Please use Prosopo Errors from common. There's a few error types, use the one which is most relevant.

Reply:
throw new ProsopoContractError('CAPTCHA.INVALID_BLOCK_NO', { context: { error: 'Block number not found' } })

Reply:
I am going to add this